File: easypermissions/src/main/java/com/almosting/easypermissions/helper/SupportFragmentPermissionHelper.java
package com.almosting.easypermissions.helper;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;

/**
 * Created by w.feng on 2018/10/10
 * Email: fengweisb@gmail.com
 */
public class SupportFragmentPermissionHelper extends BaseSupportPermissionHelper<Fragment> {
  public SupportFragmentPermissionHelper(@NonNull Fragment host) {
    super(host);
  }

  @Override
  public FragmentManager getSupportFragmentManager() {
    return getHost().getChildFragmentManager();
  }

  @Override
  public void directRequestPermissions(int requestCode, @NonNull String... perms) {
    getHost().requestPermissions(perms, requestCode);
  }

  @Override
  public boolean shouldShowRequestPermissionRationale(@NonNull String perm) {
    return getHost().shouldShowRequestPermissionRationale(perm);
  }

  @Override
  public Context getContext() {
    return getHost().getActivity();
  }
}

File: easypermissions/src/main/java/com/almosting/easypermissions/helper/SupportFragmentPermissionHelper.kt
package com.almosting.easypermissions.helper

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

/**
 * Created by w.feng on 2018/10/10
 * Email: fengweisb@gmail.com
 */
class SupportFragmentPermissionHelper(host: Fragment) :
  BaseSupportPermissionHelper<Fragment>(host) {
  override val supportFragmentManager: FragmentManager
    get() = host.childFragmentManager

  override fun directRequestPermissions(
    requestCode: Int,
    vararg perms: String
  ) {
    host.requestPermissions(perms, requestCode)
  }

  override fun shouldShowRequestPermissionRationale(perm: String): Boolean {
    return host.shouldShowRequestPermissionRationale(perm)
  }

  override val context: Context?
    get() = host.activity
}